ast-checks/src/missing-patch-comment.rs

```diff
@@ -0,0 +1,102 @@
+use codespan::{FileId, Files};
+use comment_finders::{find_comment_above, find_comment_within};
+use rnix::types::*;
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, env, error::Error, fs, process};
+use tree_utils::walk_keyvalues_filter_key;
+
+mod comment_finders;
+mod tree_utils;
+
+#[derive(Serialize, Deserialize, Debug)]
+struct SourceLocation {
+    column: usize,
+    line: usize,
+    file: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct MissingCommentMessage {
+    cond: bool,
+    locations: Vec<SourceLocation>,
+    msg: &'static str,
+    name: &'static str,
+}
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let args: Vec<String> = env::args().skip(1).collect();
+    let mut report = HashMap::new();
+
+    let mut files = Files::new();
+    for filename in args {
+        let file_id = files.add(filename.clone(), fs::read_to_string(&filename)?);
+        report.insert(filename.clone(), analyze_single_file(&files, file_id)?);
+    }
+
+    println!("{}", serde_json::to_string(&report)?);
+    Ok(())
+}
+
+fn analyze_single_file(
+    files: &Files<String>,
+    file_id: FileId,
+) -> Result<Vec<MissingCommentMessage>, Box<dyn Error>> {
+    let ast = rnix::parse(files.source(file_id)).as_result()?;
+    let root = ast.root().inner().ok_or("No elements")?;
```

Comment:
Can we make the error message clearer. Ideally something like `"No elements in the AST for attribute {} in path {}"`